plaidml/edsl/edsl_test.py

```diff
@@ -1034,6 +1034,86 @@ def test_two_outputs(self):
         self.assertEqual(outputs[0].tolist(), [1, 2, 3])
         self.assertEqual(outputs[1].tolist(), [1, 2, 3])
 
+    def test_placeholder_dtype_only(self):
+        I = Placeholder(plaidml.DType.INT32)
+        program = Program('placeholder_dtype_only', [I])
+        expected = '''
+!i32 = type tensor<!eltwise.i32>
+module {
+  func @placeholder_dtype_only(%arg0: !i32) -> !i32 {
+    %0 = "eltwise.ident"(%arg0) : (!i32) -> !i32
+    return %0 : !i32
+  }
+}
+'''
+        self.compare_results(program, expected)
```

Comment:
This is actually pretty brittle. I know we have lots of tests like this, but I don't think we should add anymore of them. So let's just create the `Program` and leave it at that.

Reply:
Maybe we could do a substring or regex comparison so it's not so brittle, so that we are testing only the bits we care about from the `program.str()`

Reply:
This might not be perfect, but I can pair up each case with the identity `LogicalShape` placeholder and compare those, i.e.

```
    def test_placeholder_with_name(self):
        I1 = Placeholder(plaidml.DType.INT32, [1, 1], name='I')
        program1 = Program('placeholder_with_name', [I1])
        I2 = Placeholder(LogicalShape(plaidml.DType.INT32, [1, 1]), name='I')
        program2 = Program('placeholder_with_name', [I2])
        self.assertEqual(str(I1.shape), "tensor<1x1x!eltwise.i32>")
        self.assertEqual(str(I2.shape), "tensor<1x1x!eltwise.i32>")
        self.compare_results(program1, str(program2))
```

Reply:
OK, comparing shapes seems good, but you can drop the `self.compare_results`

Reply:
Actually, if we're comparing shapes, then there's probably no need to even make a program.